primitiveFTPd/src/org/primftpd/filesystem/SafSshFile.java

```diff
@@ -76,7 +74,7 @@ public SshFile getParentFile() {
             parentPath = "/";
         }
         logger.trace("[{}]   getParentFile() -> {}", name, parentPath);
-        return fileSystemView.getFile(parentPath);
+        return (SshFile)fileSystemView.getFile(parentPath);
```

Comment:
why is this cast necessary? why was it not before?

Reply:
fileSystemView was moved from SafSshFile up to SafFile to give access to the getTimeResolution() method for SafFtpFile.

And that's why it's type was changed from SafSshFileSystemView to SafFileSystemView, so getFile() has lost it's return type from SshFile to... T = SafFile<SshFile>. I've tried to explain java, that T is not only SafFile<SshFile>, but also implements SshFile, but couldn't find a way. I'm new to java, so I'm not sure whether it is possible at all, but it seems to be not.

Reply:
And finally only the timeResolution is stored in SafFile, because only this is needed, so I've deleted most of my previous changes.